src/rust/ide/src/view/layout.rs

```diff
@@ -106,10 +106,10 @@ impl ViewLayout {
         let logger        = Logger::sub(logger,"ViewLayout");
         let world         = &application.display;
         let text_editor   = TextEditor::new(&logger,world,text_controller,kb_actions,fonts);
-        let node_editor   = NodeEditor::new(&logger,application,graph_controller,project,
-            visualization_controller);
+        let node_editor   = NodeEditor::new(&logger,application,graph_controller,
+            project.clone_ref(),visualization_controller);
```

Comment:
Why do we have an API which requires clone ref on the call side? This sohuld accept &project instead.

Reply:
No, because it takes project. If it would take reference, it would have to clone_ref it anyway, and now it sometimes can omit cloning if the user does not need the `project` anymore.

Reply:
Then why it doesnt accept `Into<..>` ? Then you can pass reference instead of requiring the user to clone it on call side. If this is CloneRefable, then `Into` is implemented for its reference by default.

Reply:
Hmm, I could, however for me it's a minimal difference if I call &project, and project.clone_ref() - I need to put some info anyway, and now you more clearly see what is happening during call. I usually follow the Rust philosophy telling that "explicit is better than implicit".

Another argument: Why converting anything should be responsibility of "new" function, if it's clear responsibility is just to create the `NodeEditor`?